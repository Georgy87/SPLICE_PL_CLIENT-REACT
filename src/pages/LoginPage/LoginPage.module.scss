@import '../../styles/variables.scss';

.authBox {
	display: flex;
	flex-direction: column;
	border-radius: 8px;
	justify-content: center;
	background-color: rgb(0, 0, 0);
	background-color: rgba(0, 0, 0, 0.4);
	position: absolute;
	z-index: 2;
	width: 400px;
	padding: 20px;
	text-align: center;
	.authLabel {
		display: flex;
		justify-content: start;
		margin: 0 auto;
		border-bottom: 2px solid rgb(101, 215, 243);
		margin-bottom: 40px;
		width: 240px;
		h1 {
			text-align: center;
			font-weight: 500;
			color: #fff;
			font-size: 17px;
		}
	}
}

.textbox {
	display: flex;
	justify-content: start;
    flex-direction: column;
	border: none;
	background: none;
	height: 58px;
    padding-left: 40px;
    padding-right: 40px;
	input {
		border: none;
		height: 10px;
		border-bottom: 1px solid #fff;
		background-color: transparent;
		color: #fff;
		padding: 10px;
		&:active,
		&:hover,
		&:focus {
			color: #fff;
			outline: 0;
			outline-offset: 0;
		}
	}
    p {
        color: $error;
        font-size: 13px;
        font-weight: 300;
    }
	input:-webkit-autofill,
	input:-webkit-autofill:hover,
	input:-webkit-autofill:focus,
	input:-webkit-autofill:active {
		-webkit-transition: 'color 9999s ease-out, background-color 9999s ease-out';
		-webkit-transition-delay: 9999s;
	}
}

.formFooter {
	display: flex;
	align-items: center;
	margin: 0 auto;
	margin-top: 10px;
	width: 300px;
	p {
		font-size: 12px;
		color: #fff;
	}
	a {
		cursor: pointer;
		margin-left: 5px;
		font-size: 13px;
		font-family: 500;
		color: rgb(101, 215, 243);
	}
}
