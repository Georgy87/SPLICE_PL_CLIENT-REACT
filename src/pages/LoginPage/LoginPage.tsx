import React, { useEffect, useState } from 'react';
import { useForm } from 'react-hook-form';
import { yupResolver } from '@hookform/resolvers/yup';
import { NavLink, useNavigate } from 'react-router-dom';
import { useSelector } from 'react-redux';

import { ButtonLayout } from '../../layouts/ButtonLayout/ButtonLayout';
import { AuthorizationLayout } from '../../layouts/AuthorizationLayout/AuthorizationLayout';
import { selectAuth, selectErrorMessage } from '../../store/selectors/userSelectors';
import { LoginFormSchema } from '../../utils/useFormSchemas';
import { fetchLogin } from '../../store/slices/user/actions';
import { useAppDispatch } from '../../store/types';

import styles from './LoginPage.module.scss';

export type FormProps = {
    email: string;
    password: string;
};

export const LoginPage = () => {
    const auth = useSelector(selectAuth);
    const errorMessage = useSelector(selectErrorMessage);

    const [email, setEmail] = useState<string>('');
    const [password, setPassword] = useState<string>('');

    const navigate = useNavigate();

    const onChangeEmail = (e: React.ChangeEvent<HTMLInputElement>) => {
        setEmail(e.target.value);
    };

    const onChangePassword = (e: React.ChangeEvent<HTMLInputElement>) => {
        setPassword(e.target.value);
    };

    const {
        register,
        handleSubmit,
        formState: { errors },
        reset,
    } = useForm<FormProps>({
        resolver: yupResolver(LoginFormSchema),
    });

    useEffect(() => {
        if (auth) {
            navigate('/');
        }
    }, [auth]);

    const dispatch = useAppDispatch();

    const onSubmit = (data: any) => {
        dispatch(fetchLogin(data));
        reset();
    };

    return (
        <div data-testid="login-page">
            <AuthorizationLayout>
                <form className={styles.authBox} onSubmit={handleSubmit(onSubmit)}>
                    <div className={styles.authLabel}>
                        <h1>Log In to Your Splice Account!</h1>
                    </div>
                    <div className={styles.textbox}>
                        <input
                            type="email"
                            placeholder="Email"
                            {...register('email')}
                            value={email}
                            onChange={(e: React.ChangeEvent<HTMLInputElement>) => onChangeEmail(e)}
                            data-testid="login_input"
                        />
                        <p data-testid="error-valid-message">{errors.email?.message}</p>
                    </div>
                    <div className={styles.textbox}>
                        <input
                            type="password"
                            placeholder="Password"
                            {...register('password')}
                            value={password}
                            onChange={(e: React.ChangeEvent<HTMLInputElement>) => onChangePassword(e)}
                            data-testid="password_input"
                        />
                        <p>{errorMessage ? errorMessage : errors.password?.message}</p>
                    </div>
                    <ButtonLayout typeStyle='auth'>Log In</ButtonLayout>
                    <div className={styles.formFooter}>
                        <p>Don't have an account?</p>
                        <NavLink to="/registration" data-testid="registration-link">
                            Sign up?
                        </NavLink>
                    </div>
                </form>
            </AuthorizationLayout>
        </div>
    );
};
