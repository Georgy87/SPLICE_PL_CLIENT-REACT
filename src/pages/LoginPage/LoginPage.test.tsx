import { render } from '@testing-library/react';
import { renderWithRouter } from '../../tests/helpers/RenderWithRouter';

import { LoginPage } from './LoginPage';

const setUp = () => shallow()
describe('LoginPage component', () => {
	test('LoginPage snapshot', async () => {
		const loginPage = render(renderWithRouter(<LoginPage />, '/login'));
		expect(loginPage).toMatchSnapshot();
	});
});
