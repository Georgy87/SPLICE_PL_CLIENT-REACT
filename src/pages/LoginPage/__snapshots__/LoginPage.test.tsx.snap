// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component App snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        data-testid="login-page"
      >
        <div
          class="root"
        >
          <div
            class="bgAuthImage"
          />
          <form
            class="authBox"
          >
            <div
              class="authLabel"
            >
              <h1>
                Log In to Your Splice Account!
              </h1>
            </div>
            <div
              class="textbox"
            >
              <input
                name="email"
                placeholder="Email"
                type="email"
                value=""
              />
              <p />
            </div>
            <div
              class="textbox"
            >
              <input
                name="password"
                placeholder="Password"
                type="password"
                value=""
              />
              <p />
            </div>
            <div
              class="platformBtn"
            >
              <button
                class="platformBtn auth"
                type="submit"
              >
                Log In
              </button>
            </div>
            <div
              class="formFooter"
            >
              <p>
                Don't have an account?
              </p>
              <a
                class=""
                data-testid="registration-link"
                href="/registration"
              >
                Sign up?
              </a>
            </div>
          </form>
        </div>
      </div>
      <div
        data-testid="login-page"
      >
        <div
          class="root"
        >
          <div
            class="bgAuthImage"
          />
          <form
            class="authBox"
          >
            <div
              class="authLabel"
            >
              <h1>
                Log In to Your Splice Account!
              </h1>
            </div>
            <div
              class="textbox"
            >
              <input
                name="email"
                placeholder="Email"
                type="email"
                value=""
              />
              <p />
            </div>
            <div
              class="textbox"
            >
              <input
                name="password"
                placeholder="Password"
                type="password"
                value=""
              />
              <p />
            </div>
            <div
              class="platformBtn"
            >
              <button
                class="platformBtn auth"
                type="submit"
              >
                Log In
              </button>
            </div>
            <div
              class="formFooter"
            >
              <p>
                Don't have an account?
              </p>
              <a
                class=""
                data-testid="registration-link"
                href="/registration"
              >
                Sign up?
              </a>
            </div>
          </form>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      data-testid="login-page"
    >
      <div
        class="root"
      >
        <div
          class="bgAuthImage"
        />
        <form
          class="authBox"
        >
          <div
            class="authLabel"
          >
            <h1>
              Log In to Your Splice Account!
            </h1>
          </div>
          <div
            class="textbox"
          >
            <input
              name="email"
              placeholder="Email"
              type="email"
              value=""
            />
            <p />
          </div>
          <div
            class="textbox"
          >
            <input
              name="password"
              placeholder="Password"
              type="password"
              value=""
            />
            <p />
          </div>
          <div
            class="platformBtn"
          >
            <button
              class="platformBtn auth"
              type="submit"
            >
              Log In
            </button>
          </div>
          <div
            class="formFooter"
          >
            <p>
              Don't have an account?
            </p>
            <a
              class=""
              data-testid="registration-link"
              href="/registration"
            >
              Sign up?
            </a>
          </div>
        </form>
      </div>
    </div>
    <div
      data-testid="login-page"
    >
      <div
        class="root"
      >
        <div
          class="bgAuthImage"
        />
        <form
          class="authBox"
        >
          <div
            class="authLabel"
          >
            <h1>
              Log In to Your Splice Account!
            </h1>
          </div>
          <div
            class="textbox"
          >
            <input
              name="email"
              placeholder="Email"
              type="email"
              value=""
            />
            <p />
          </div>
          <div
            class="textbox"
          >
            <input
              name="password"
              placeholder="Password"
              type="password"
              value=""
            />
            <p />
          </div>
          <div
            class="platformBtn"
          >
            <button
              class="platformBtn auth"
              type="submit"
            >
              Log In
            </button>
          </div>
          <div
            class="formFooter"
          >
            <p>
              Don't have an account?
            </p>
            <a
              class=""
              data-testid="registration-link"
              href="/registration"
            >
              Sign up?
            </a>
          </div>
        </form>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`LoginPage component LoginPage snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        data-testid="login-page"
      >
        <div
          class="root"
        >
          <div
            class="bgAuthImage"
          />
          <form
            class="authBox"
          >
            <div
              class="authLabel"
            >
              <h1>
                Log In to Your Splice Account!
              </h1>
            </div>
            <div
              class="textbox"
            >
              <input
                name="email"
                placeholder="Email"
                type="email"
                value=""
              />
              <p />
            </div>
            <div
              class="textbox"
            >
              <input
                name="password"
                placeholder="Password"
                type="password"
                value=""
              />
              <p />
            </div>
            <div
              class="platformBtn"
            >
              <button
                class="platformBtn auth"
                type="submit"
              >
                Log In
              </button>
            </div>
            <div
              class="formFooter"
            >
              <p>
                Don't have an account?
              </p>
              <a
                class=""
                data-testid="registration-link"
                href="/registration"
              >
                Sign up?
              </a>
            </div>
          </form>
        </div>
      </div>
      <div
        data-testid="login-page"
      >
        <div
          class="root"
        >
          <div
            class="bgAuthImage"
          />
          <form
            class="authBox"
          >
            <div
              class="authLabel"
            >
              <h1>
                Log In to Your Splice Account!
              </h1>
            </div>
            <div
              class="textbox"
            >
              <input
                name="email"
                placeholder="Email"
                type="email"
                value=""
              />
              <p />
            </div>
            <div
              class="textbox"
            >
              <input
                name="password"
                placeholder="Password"
                type="password"
                value=""
              />
              <p />
            </div>
            <div
              class="platformBtn"
            >
              <button
                class="platformBtn auth"
                type="submit"
              >
                Log In
              </button>
            </div>
            <div
              class="formFooter"
            >
              <p>
                Don't have an account?
              </p>
              <a
                class=""
                data-testid="registration-link"
                href="/registration"
              >
                Sign up?
              </a>
            </div>
          </form>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      data-testid="login-page"
    >
      <div
        class="root"
      >
        <div
          class="bgAuthImage"
        />
        <form
          class="authBox"
        >
          <div
            class="authLabel"
          >
            <h1>
              Log In to Your Splice Account!
            </h1>
          </div>
          <div
            class="textbox"
          >
            <input
              name="email"
              placeholder="Email"
              type="email"
              value=""
            />
            <p />
          </div>
          <div
            class="textbox"
          >
            <input
              name="password"
              placeholder="Password"
              type="password"
              value=""
            />
            <p />
          </div>
          <div
            class="platformBtn"
          >
            <button
              class="platformBtn auth"
              type="submit"
            >
              Log In
            </button>
          </div>
          <div
            class="formFooter"
          >
            <p>
              Don't have an account?
            </p>
            <a
              class=""
              data-testid="registration-link"
              href="/registration"
            >
              Sign up?
            </a>
          </div>
        </form>
      </div>
    </div>
    <div
      data-testid="login-page"
    >
      <div
        class="root"
      >
        <div
          class="bgAuthImage"
        />
        <form
          class="authBox"
        >
          <div
            class="authLabel"
          >
            <h1>
              Log In to Your Splice Account!
            </h1>
          </div>
          <div
            class="textbox"
          >
            <input
              name="email"
              placeholder="Email"
              type="email"
              value=""
            />
            <p />
          </div>
          <div
            class="textbox"
          >
            <input
              name="password"
              placeholder="Password"
              type="password"
              value=""
            />
            <p />
          </div>
          <div
            class="platformBtn"
          >
            <button
              class="platformBtn auth"
              type="submit"
            >
              Log In
            </button>
          </div>
          <div
            class="formFooter"
          >
            <p>
              Don't have an account?
            </p>
            <a
              class=""
              data-testid="registration-link"
              href="/registration"
            >
              Sign up?
            </a>
          </div>
        </form>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
